elide-core/src/main/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialect.java

```diff
@@ -65,7 +68,8 @@
     private static final String SINGLE_PARAMETER_ONLY = "There can only be a single filter query parameter";
     private static final String INVALID_QUERY_PARAMETER = "Invalid query parameter: ";
     private static final Pattern TYPED_FILTER_PATTERN = Pattern.compile("filter\\[([^\\]]+)\\]");
-    private static final Pattern FILTER_ARGUMENTS_PATTERN = Pattern.compile("\\[(.+?)\\]");
+    // field name followed by zero or more square brackets having non-empty argument name and value separated by ':'
+    private static final String FILTER_SELECTOR_REGEX = "(\\w+)(\\[(\\w+?):([^\\]]+?)\\])*$";
```

Comment:
Do we need the ? marks here?

Reply:
it will work correctly without `?` also so removing them.